lib/msf/core/post/file.rb

```diff
@@ -501,19 +501,25 @@ def rm_rf(*remote_dirs)
   alias :dir_rm :rm_rf
 
   #
-  # Rename a remote file.
+  # Renames a remote file and returns true on success and false
+  # on failure
   #
   # @param old_file [String] Remote file name to move
   # @param new_file [String] The new name for the remote file
   def rename_file(old_file, new_file)
+    unless exist?(old_file)
+      elog("The file to be renamed does not exists!")
+      return false
+    end
+    verification_token = Rex::Text.rand_text_alphanumeric()
     if session.type == "meterpreter"
       return (session.fs.file.mv(old_file, new_file).result == 0)
+    elsif session.type == 'powershell'
+      !!cmd_exec("Rename-Item -Path \"#{old_file}\" -NewName \"#{new_file}\" && echo #{verification_token}") =~ /#{verification_token}/
+    elsif session.platform == 'windows'
+      !!(cmd_exec(%Q|move /y "#{old_file}" "#{new_file}" & if not errorlevel 1 echo #{verification_token}|) =~ /#{verification_token}/)
     else
-      if session.platform == 'windows'
-        cmd_exec(%Q|move /y "#{old_file}" "#{new_file}"|) =~ /moved/
-      else
-        cmd_exec(%Q|mv -f "#{old_file}" "#{new_file}"|).empty?
-      end
+      !!cmd_exec(%Q|mv -f "#{old_file}" "#{new_file}" && echo #{verification_token}|).strip =~ /verification_token/
```

Comment:
This is checking for the literal string `verification_token`.

Reply:
okay!